src/gui/components.py

```diff
@@ -28,23 +31,116 @@ def add_qrcode(text, y=QR_PADDING, scr=None, style=None, width=None):
     qr.set_size(width)
     qr.set_text(text)
     qr.align(scr, lv.ALIGN.IN_TOP_MID, 0, y)
+
     return qr
 
-class QRCode(lv.label):
+class QRAnim:
+    """
+    Example of animated qr: p1of3 payload
+    """
+    def __init__(self):
+        # indx = [0, 2] means first part out of 3
+        self.indx = []
+        # list of payloads for every QR part
+        self.payload = []
+        # animated qr code must have space for buttons
+        self.MAX_WIDTH = const(350)
+
+    def header(self):
+        if len(self.indx) != 2:
+            return ""
+        return "p" + str(self.indx[0]+1) + "of" + str(self.indx[1]+1) + " "
+
+    def indx_to_str(self):
+        return "Part: " + str(self.indx[0]+1) + " / " + str(self.indx[1]+1)
+
+    @staticmethod
+    def average_chunk_size(msg_size, max_chunk_size):
+        number_of_chunks = ceil(msg_size / max_chunk_size)
+        return ceil(msg_size / number_of_chunks)
+
+    def pending(self, text = None):
+        if text == None:
+            return self.indx != []
+        return text in self.payload or text == "".join(self.payload)
+
+    def clean(self):
+        self.indx = []
+        self.payload = []
+
+class QRCode(lv.cont):
+    # number of bytes to be QR encoded
+    MAX_SIZE = const(300) # FIXME
```

Comment:
This should be large enough so that we dont unnecessarily animate other stuff, e.g. MASTER SINGLE KEY animation will cause overlapping of the message with the show-derivation-path slider. Also, small animations look a bit different (tight) due to a bit unpredictable QR padding algorithm